### DISCUSSION TITLE
Filter By Date/Filter data between dates

### DISCUSSION THREAD

**Discussion by Mr-Mohammad** [alirezanet/Gridify#148] [Q&A] [answered]
Hi guys,
I want to make a query between two dates, but unfortunately the result returns null for me
My Filter: CreateDate>=2021-01-01,CreateDate<=2024-01-01
Type Of CreateDate : DateTime

**Comment by alirezanet** [comment]
Hi @Mr-Mohammad,
Did you map your date properties correctly? If you share your mappings I could help you better.

**Reply by Mr-Mohammad** [reply]
var mapper = new GridifyMapper<Users>(true)
                .AddMap("FirstName", v => v.FirstName)
                .AddMap("LastName", v => v.LastName)
                .AddMap("NationalCode", v => v.NationalCode)
                .AddMap("PhoneNumber", v => v.PhoneNumber)
                .AddMap("UserName", v => v.UserName)
                .AddMap("IsActive", v => v.IsActive)
                .AddMap("Gender", v => v.Gender)
                .AddMap("CreateDate", v=>v.CreateDate);

            var res = _dbContext.Users.Gridify(query, mapper);

**Reply by alirezanet** [reply]
I think you're trying to compare dates but you've forgotten to exclude the `Time`, so changing the mapping to `v.CreateDate.Date` should give you what you need. 

e.g:
```csharp
 var lst = new List<Test>()
{
	new(DateTime.Parse("2024-01-01 14:11:23")),
	new(DateTime.Parse("2024-01-02 11:12:53 ")),
	new(DateTime.Parse("2024-01-03 10:11:23"))
};

var qb = new QueryBuilder<Test>()
	.AddMap(nameof(Test.CreateDate), q=> q.CreateDate.Date)
	.AddCondition("createDate<=2024-01-02")
	.Build(lst);


public record Test(DateTime CreateDate);
```

**Reply by alirezanet** [reply, marked as answer]
also, I should note these mappings are not required since you're using the same property names.
```
          .AddMap("FirstName", v => v.FirstName)
            .AddMap("LastName", v => v.LastName)
            .AddMap("NationalCode", v => v.NationalCode)
            .AddMap("PhoneNumber", v => v.PhoneNumber)
            .AddMap("UserName", v => v.UserName)
            .AddMap("IsActive", v => v.IsActive)
            .AddMap("Gender", v => v.Gender)
```
So you need just this part:
```csharp
 var mapper = new GridifyMapper<Users>(true)
            .AddMap("CreateDate", v=>v.CreateDate.Date);

 var res = _dbContext.Users.Gridify(query, mapper);
```
